librz/util/file.c

```diff
@@ -1257,3 +1257,61 @@ RZ_API char *rz_file_path_join(const char *s1, const char *s2) {
 	const char *sep = ends_with_dir ? "" : RZ_SYS_DIR;
 	return rz_str_newf("%s%s%s", s1, sep, s2);
 }
+
+/**
+ * \brief zip the contents of src and store in dst
+ * \param src source file (string containing filename)
+ * \param dst destination file (string containing filename)
+ * \return true, if successful; false otherwise
+ */
+RZ_API bool rz_file_deflate(RZ_NONNULL const char *src, RZ_NONNULL const char *dst) {
```

Comment:
I think the implementation of this function is good enough as a first implementaiton, but it requires loading the whole file in memory which imo is not ideal. I believe the zlib library has a stream API by default, so you can call inflate/deflate by providing more data as necessary. If that is so, we could use `rz_buf_new_file` to create a `RzBuffer` from the `src` file and then have a `rz_deflate_buf` function that uses `rz_buf_read` as necessary to read data from the buffer. In this way we don't need to load the whole file in memory. What do you think?

Reply:
It doesn't work as expected, as of now. I will first need to fix that. After that, I can definitely work on using streaming behaviour instead of storing the whole file in memory.

Reply:
I have implemented this in my last commit. Please take a look. However, this breaks `rz_file_inflate`. I don't know of any way to have streaming behaviour in `rz_file_inflate`, and just using the current implementation gives `RZ_ERR_INVALID_CONTENTS` error. However, when I manually decompress (using `gzip`) the file generated by `rz_file_inflate` and open that with rizin, it works perfectly fine.

Reply:
I propose to add the APIs:
```C
RZ_API RzBuffer *rz_inflate_buf(RzBuffer *src);
RZ_API RzBuffer *rz_deflate_buf(RzBuffer *src);
```